### PR TITLE
Added Lightrun and debugging/APM section

### PR DESCRIPTION
**Pull request by shai-almog**

### CODE REVIEW

**Comment by agamm** [comment]
I would add this to monitoring or open a new debugging category with competitors (I don't like to open a category without any other tools). I see there is also Rookout in the same space.

**Comment by shai-almog** [comment]
Good point. I'm not familiar enough with Ozcode to write anything about it and their website is a bit ambiguous about the technology so I left it out. But I do know the google cloud solution and rookout which I added.

File: README.md [hunk 1]
```diff
@@ -85,6 +85,12 @@ The definition of developer-first for this repo is:
 *Storing data and processing it.*
 * [Airtable](https://airtable.com/) - Database and spreadsheet hybrid with an API.
 
+## Debugging 
+*Tools that help in tracking bugs.*
+* [Google Cloud's operations suite](https://cloud.google.com/products/operations) -  Set of debug primitives (capture, log injection etc) for Google cloud
```
Comment (agamm) [inline README.md:90]:
```suggestion
* [Google Cloud's operations suite](https://cloud.google.com/products/operations) -  Set of debug primitives (capture, log injection, etc) for Google cloud.
```

Comment (agamm) [reply README.md:90]: Are you sure this is a widely known solution? I mean if people use this they don't need to use the other ones? plus it only seems to be relevant to GCP, no?

File: README.md [hunk 2]
```diff
@@ -85,6 +85,12 @@ The definition of developer-first for this repo is:
...
+* [Lightrun](https://www.lightrun.com) - Debug production servers using a plugin in the IDE (on prem or SaaS) supports Java, Node.js and Python
```
Comment (agamm) [inline README.md:91]:
```suggestion
* [Lightrun](https://www.lightrun.com) - Debug production servers using a plugin in the IDE (on-prem or SaaS) that supports Java, Node.js, and Python.
```

File: README.md [hunk 3]
```diff
@@ -85,6 +85,12 @@ The definition of developer-first for this repo is:
...
+* [Rookout](https://www.rookout.com) - Debug production servers via cloud based debugger. Supports Java, .Net, Node.js, Python, and Ruby
```
Comment (agamm) [inline README.md:92]:
```suggestion
* [Rookout](https://www.rookout.com) - Debug production servers via a cloud-based debugger. Supporting Java, .Net, Node.js, Python, and Ruby.
```

**Review by agamm — CHANGES_REQUESTED** [review]
Added some grammar fixes and a question about the GCP solution.

File: README.md (hunk 3 above)
Comment (agamm) [reply README.md:92]: Does rookout also support on-prem in some way? what is the other difference from lightrun? (to be consistent with the "on-prem or SaaS" you wrote :)

Comment (shai-almog) [reply README.md:92]: Rookout supports more platforms and has been around longer. It's a cloud based solution so while it does have an IDE plugin it's main focus is on debugging via the web interface.  It doesn't have an on-prem option but it does have a locally installed proxy option. I'm not sure if that can be considered on-prem.

The three tools are conceptually similar. They try to provide the power of debugging local applications on a production server securely. Their approaches are also similar: agent installed on your server connected to their backend. The differences are:

- Agent implementations - this varies a bit but is too subjective to quantify so not relevant here
- Languages supported
- Vendor specific - Google only runs on GCP as far as I know
- IDE integration vs. Web App
- On-prem vs. SaaS - AFAIK only Lightun offers an on-prem option

Comment (agamm) [reply README.md:92]: Awesome, thanks for the clarification. Let's commit the grammar changes and merge?

File: README.md (hunk 1 above)
Comment (shai-almog) [reply README.md:90]: The GCP option is pretty widely known. It's mostly known as stackdriver but Google renamed it recently.

File: README.md (hunk 3 above)
Comment (shai-almog) [reply README.md]: Sorry I thought I committed that, should be done now.

**Review by agamm — APPROVED** [review]
Awesome, thanks for the participation and responsiveness :)

**Pull request merged**